src/pages/index.js

```diff
@@ -78,13 +149,55 @@ const addFormVaildator = new FormValidator(
 );
 addFormVaildator.enableValidation();
 
+const avatarFormValidator = new FormValidator(
+  validationSettings,
+  constants.updateAvatarForm
+);
+avatarFormValidator.enableValidation();
+
 // ! ||--------------------------------------------------------------------------------||
 // ! ||                                   Functions                                    ||
 // ! ||--------------------------------------------------------------------------------||
 
+function handleDeleteCardClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.submitButton.textContent = "Saving...";
+    api
+      .removeCard(card.getId())
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+        alert(`${err}, Could not delete card`);
+      })
+      .finally(() => {
+        deleteCardPopup.submitButton.textContent = "Yes";
+      });
+  });
+}
+
 function renderCard(item) {
-  const cardEl = new Card(item, selectors.cardTemplate, handleImageClick);
-  cardSection.addItem(cardEl.getView());
+  const card = new Card(
+    item,
+    selectors.cardTemplate,
+    handleImageClick,
+    handleDeleteCardClick,
+    () => {
+      if (!card.isLiked) {
+        api.addLike(card._id).then((res) => {
+          card.setIsLiked(res.isLiked);
+        });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere**